src/app/modules/truenas-connect/components/truenas-connect-status-modal/truenas-connect-status-modal.component.ts

```diff
@@ -64,21 +69,40 @@ export class TruenasConnectStatusModalComponent {
 
   constructor(
     @Inject(WINDOW) private window: Window,
-    private matDialog: MatDialog,
     protected tnc: TruenasConnectService,
   ) { }
 
-  protected openSettings(): void {
-    this.matDialog
-      .open(TruenasConnectModalComponent, {
-        width: '456px',
-      })
-      .afterClosed()
-      .pipe(untilDestroyed(this))
-      .subscribe();
+  ngOnInit(): void {
+    // Automatically re-enable the service if it's disabled
```

Comment:
Instead of a comment, we can have a method here that describes what we want to do. Something like
`enableServiceIfDisabled()` and move this code inside the method.